### PR TITLE
[Select] hide QueryList item list when menuContent and createItemView are both null

### PR DESCRIPTION
**Pull request by tnrich**
Links https://github.com/palantir/blueprint/issues/3425

### CODE REVIEW

**Comment by tnrich** [comment]
I'm not sure why this test would be failing: 
```
<TagInput> onAdd is invoked on blur when addOnBlur=true
```
?

File: packages/select/src/components/query-list/queryList.tsx [hunk 1]
```diff
@@ -262,6 +262,9 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
         const maybeNoResults = this.isCreateItemRendered() ? null : noResults;
         const menuContent = renderFilteredItems(listProps, maybeNoResults, initialContent);
         const createItemView = this.isCreateItemRendered() ? this.renderCreateItemMenuItem(this.state.query) : null;
+        if (menuContent === null && createItemView === null) {
```
Comment (adidahiya) [inline packages/select/src/components/query-list/queryList.tsx]: I think you should check `== null` in both cases. we don't have strict null checks enabled in this project. `maybeNoResults` could be `undefined` at this point.

**Comment by adidahiya** [comment]
Don't worry about the test flake, I'll look into it separately

File: packages/select/src/components/query-list/queryList.tsx (hunk 1 above)
Comment (giladgray) [reply packages/select/src/components/query-list/queryList.tsx]: yes `==` is essential here. looks good otherwise.

**Comment by tnrich** [comment]
I've changed those equality checks as suggested. Thanks!

**Comment by tnrich** [comment]
whoohoo!

**Pull request merged**